### PR TITLE
Issue #3907: Added period at the end of first sentence

### PR DESCRIPTION
**Pull request by sagar-shah94**
Issue #3907

### CODE REVIEW

**Comment by sagar-shah94** [comment]
@romani as per your instructions i have created separate PR.

File: src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java [hunk 1]
```diff
@@ -32,8 +32,8 @@
 import com.puppycrawl.tools.checkstyle.utils.ModuleReflectionUtils;
 
 /**
- * A factory for creating objects from package names and names,
- * considering:
+ * A factory for creating objects from package names and names.
+ * consider below example for better understanding.
```
Comment (MEZk) [inline src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java]: The first word after dot should start with a capital letter.

Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java]: `below` => `the below`

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/AbstractTypeAwareCheck.java [hunk 2]
```diff
@@ -209,6 +209,7 @@ protected static boolean isSubclass(Class<?> child, Class<?> parent) {
     }
 
     /**
+     * If classresolver is not initialised then it will be initialised and returned.
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/AbstractTypeAwareCheck.java]: First sentence should say what the method does, not any of it's internal workings.
Please change it to something like `Returns the current tree's ClassResolver.` or something else that doesn't imitate the `return` description so much.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/ClassResolver.java [hunk 3]
```diff
@@ -184,6 +184,7 @@ public ClassResolver(ClassLoader loader, String pkg, Set<String> imports) {
     }
 
     /**
+     * Returns true if specified class is loadable with safeLoad() or false.
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/ClassResolver.java]: Again, describe what the method does. What you have written could just go directly in the `@return` description.
Change to something like `Checks if the given class name can be loaded.`.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java [hunk 4]
```diff
@@ -166,6 +166,7 @@ else if (!hasText(leftCurly)) {
     }
 
     /**
+     * Returns true if SLIST token contains any text or else false.
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java]: Same here as previous.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java [hunk 5]
```diff
@@ -215,7 +215,7 @@ public void visitToken(DetailAST ast) {
     }
 
     /**
-     * Checks if ast is default in annotation
+     * Checks if ast is default in annotation.
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java]: `is default in annotation` => `is the default token of an annotation field`

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ExplicitInitializationCheck.java [hunk 6]
```diff
@@ -185,6 +185,9 @@ private static boolean isNumericType(int type) {
     }
 
     /**
+     * Finds if given node contains numeric constant for zero.Returns
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ExplicitInitializationCheck.java]: Remove second sentence that just duplicates `@return`.

`Finds` => `Checks`
Find methods usually return position or Object being looked for. This method returns `boolean`.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTag.java [hunk 7]
```diff
@@ -66,6 +66,7 @@ public String getTagName() {
     }
 
     /**
+     * Returns first argument. null if not set.
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTag.java]: remove 2nd sentence.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/ParenPadCheck.java [hunk 8]
```diff
@@ -213,6 +213,7 @@ private boolean isAcceptableToken(DetailAST ast) {
     }
 
     /**
+     * Returns array of acceptableTokens.
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/ParenPadCheck.java]: `acceptableTokens` => `acceptable tokens`

**Comment by romani** [comment]
@sagar-shah94 , to resolve CIs, please rebase.

**Comment by romani** [comment]
@sagar-shah94 , please finish this PR. It is very good PR, we need to merge it.

**Comment by sagar-shah94** [comment]
@romani issue i am facing is that even after rebase ,i am getting error when i run mvn clean verify in local so stuck in that.

**Comment by romani** [comment]
@sagar-shah94 , please share all error(or whole build output) somehow with us, we will help you.

**Comment by sagar-shah94** [comment]
@rnveach @romani @MEZk  all suggested changes are made please verify and let me know if more changes need to be made to get this issue merged

**Comment by sagar-shah94** [comment]
@romani why is codecov/project CI failing .i think it shows code coverage and it is showing 99.99%.How can i make it 100% to get it merged with master?

**Comment by rnveach** [comment]
> why is codecov/project CI failing 

For whatever reason, it loses track of the coverage report from the previous commit. It happens sometimes.
Travis and code coverage is passing, so you should already have 100% code coverage (ignore grammar).

**Comment by rnveach** [comment]
@sagar-shah94 There are still quite a few method descriptions that are either duplicates or almost duplicates of the `@return` tag.
Is there any chance we can make them more unique?

@romani I have nothing else to add, we can either change the method descriptions or take it as it is.
Feel free to start reviewing.

**Comment by romani** [comment]
@rnveach , I am fine with such changes, it is far better that no comment at all. We are doing the progress.
@sagar-shah94 , thanks a lot. Please proceed with issue PR.

**Pull request merged**